core/src/main/scala/chisel3/internal/Binding.scala

```diff
@@ -28,7 +33,11 @@ object requireIsHardware {
 
 /** Requires that a node is a chisel type (not hardware, "unbound")
   */
+@deprecated(deprecatedPublicAPIMsg, "Chisel 3.6")
 object requireIsChiselType {
+  def apply(node: Data, msg: String = ""): Unit = _requireIsChiselType.apply(node, msg)
+}
+private[chisel3] object _requireIsChiselType {
   def apply(node: Data, msg: String = ""): Unit = if (node.isSynthesizable) {
     val prefix = if (msg.nonEmpty) s"$msg " else ""
     throw ExpectedChiselTypeException(s"$prefix'$node' must be a Chisel type, not hardware")
```

Comment:
This, and `requireIsHardware` above should just be moved to chisel3.experimental (where there is currently a val alias). Then, instead of objects that replicate all of the arguments, you should just put a deprecated val alias in package object chisel3.internal.